webapp/handlers.py

```diff
@@ -47,6 +47,11 @@ def cache_headers(response):
         if flask.request.path.startswith(disable_cache_on):
             response.cache_control.no_store = True
 
+        # Prevent XSS
+        if flask.request.path.startswith("/certified"):
+            response.headers["X-Frame-Options"] = "DENY"
+            response.headers["X-XSS-Protection"] = "1; mode=block"
```

Comment:
The header is not widely supported in the browsers. According to [caniuse.com](https://caniuse.com/?search=X-XSS-Protection), the browsers of only 2.9% of Internet users will be able to interpret the header.

[There is also an MDN mention that advises the developers to avoid using this feature in production websites](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection).

Please [reconsider](https://github.com/canonical/ubuntu.com/issues/1141) adding [`Content-Security-Policy`](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP), which is the modern equivalent of `X-XSS-Protection`.

Reply:
The issue with Content-Security-Policy is that it's too restrictive, I added initially but that was blocking many scripts and images, I'll see if I can tweak it.